pkgs/dart_mcp/lib/src/server/resources_support.dart

```diff
@@ -0,0 +1,168 @@
+// Copyright (c) 2025, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+part of 'server.dart';
+
+/// A mixin for MCP servers which support the `resources` capability.
+///
+/// Servers should add resources using the [addResource] method, typically
+/// inside the [initialize] method, but they may also be added after
+/// initialization if needed.
+///
+/// Resources can later be removed using [removeResource], or the client can be
+/// notified of updates using [updateResource].
+///
+/// Clients will be notified automatically of resource changes.
+///
+/// See https://modelcontextprotocol.io/docs/concepts/resources.
+mixin ResourcesSupport on MCPServer {
+  /// The current resources by URI.
+  final Map<String, Resource> _resources = {};
+
+  /// The current resource implementations by URI.
+  final Map<String, FutureOr<ReadResourceResult> Function(ReadResourceRequest)>
+  _resourceImpls = {};
+
+  /// The list of currently subscribed resources by URI.
+  final Set<String> _subscribedResources = {};
+
+  /// Invoked by the client as a part of initialization.
+  ///
+  /// Resources should usually be added in this function using [addResource]
+  /// when possible.
+  ///
+  /// If resources are added, updated, or removed after [initialized] completes,
+  /// then the client will be notified of the changes based on their
+  /// subscription preferences.
+  @override
+  FutureOr<InitializeResult> initialize(InitializeRequest request) async {
```

Comment:
should this be annotated with @mustCallSuper in case the initialize method is overriden?

Reply:
The original one is marked as such which implicitly marks all overrides the same way "In addition, every method that overrides `m` is implicitly annotated with this same annotation."